tools/Ecoregionalization_workflow/cluster.xml

```diff
@@ -0,0 +1,132 @@
+<tool id="Clara_cluster" name="ClaraClust" version="0.1.0+galaxy0">
+    <description>Make cluster from BRT prediction with Clara algorithm</description>
+    <requirements>
+       <requirement type="package" version="4.2.3">r-base</requirement>
+       <requirement type="package" version="2.1.4">r-cluster</requirement>
+       <requirement type="package" version="1.1.1">r-dplyr</requirement>
+       <requirement type="package" version="2.0.0">r-tidyverse</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        Rscript
+         '$__tool_directory__/cluster_ceamarc.R'
+         '$input1'
+         '$input2'
+         '$input3'
+         '$k'
+         '$metric'
+         '$sample'
+         '$output1'
+         '$output2'
+         '$output3'
+    ]]></command>
+    <inputs>
+      <param name="input1" type="data" format="tsv" label="Prediction matrix (data to cluster from previous step) "/>
+      <param name="input2" type="data" format="txt,csv,tsv" label="Environment file"/>
+      <param name="input3" type="data" format="tsv" label="Prediction file (data.bio table from previous step)"/>
+      <param name="k" type="integer" label="Number of Cluster wanted" value= "2"/>
+      <param name="metric" type="select" label="What metric to use to calculate dissimilarities between observations ?">
+             <option value = "manhattan">manhattan</option>
+             <option value = "euclidean">euclidean</option>
+             <option value = "jaccard">jaccard</option>
+      </param>
+      <param name="sample" type="integer" label= "The number of samples to be drawn from the dataset" value="10"/>
+    </inputs>
+    <outputs>
+      <data name="output1" from_work_dir="sih.png" format="png" label="SIH plot"/>
+      <data name="output2" from_work_dir="points_clus.txt" format="txt" label="Cluster points"/>
+      <data name="output3" from_work_dir="clus.txt" format="txt" label="Cluster info"/>
+    </outputs>
+    <tests>
+        <test>
+            <param name="input1" value="Data_to_cluster.tsv"/>
+            <param name="input2" value="ceamarc_env.csv"/>
+            <param name="input3" value="Data.bio_table.tsv"/>
+            <param name='k' value="2"/>
+            <param name='metric' value="manhattan"/>
+            <param name='sample' value="10"/>
+            <output name='output1'>
+                <assert_contents>
+            	    <has_size value="8128" delta="500"/>
+            	</assert_contents>
+            </output>
+            <output name='output2'>
+                <assert_contents>
+                    <has_size value="2452000" delta="1000" />
+                </assert_contents>
+            </output>
+            <output name='output3' >
+                <assert_contents>
+                    <has_size value="21457687" delta="1000000" />
```

Comment:
comparing the size of a text file is not a very good test, especially not with a large delta
Can you maybe check for the content or number of lines etc ... see https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output-assert-contents

Reply:
In this case, other check like number of lines is impossible because of the variability of the predictions. Indeed, this output is dependent of the number of taxa having a BRT model and if for example one more taxa obtained a model the size of this output would growth exponentially. So I can’t predict the exact output or even its size which is why the delta is large.

Reply:
But you have here fixed inputs and know how many taxa have a BRT model, do you?

Reply:
No, I don't know how many taxa have a BRT model because there are some fluctuations between runs. Some taxa will get a BRT model during some runs and not during other runs, so it’s unpredictable. That’s also why I can’t add asserts or count as you suggested in other comments.

Reply:
confirmed